site/app/views/StatsView.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+namespace app\views;
+
+use app\libraries\FileUtils;
+use app\views\Date;
+
+class StatsView extends AbstractView {
+
+    /**
+     * @param $file
+     * @param int $whatstat 0 for wait time, 1 for run time
+     * @param array $contains
+     * @param array $notcontains
+     * @return array
+     */
+
+    private function convertFileToArray($file, $whatstat = 0, $contains = array(), $notcontains = array()) {
+        $lines = explode(PHP_EOL, $file);
+        $array = array();
+        $notcontains[] = "zip";
+        $notcontains[] = "generated_output";
+        foreach ($lines as $line) {
+            $next = $line;
+            foreach ($contains as $contain) {
+                $contain = "/".$contain."/";
+                if (gettype($next) != "String" && $next == array()) {
+                    continue;
+                }
+                $next = preg_grep($contain, array($next));
+                foreach ($next as $k => $v) {
+                    $next = $v;
+                    break;
+                }
+            }
+            foreach ($notcontains as $contain) {
+                $contain = "/".$contain."/";
+                if (gettype($next) != "String" && $next == array()) {
+                    continue;
+                }
+                $next = preg_grep($contain,array($next),PREG_GREP_INVERT);
+                foreach ($next as $k=>$v) {
+                    $next = $v;
+                    break;
+                }
+            }
+            if (gettype($next) != "String" && $next == array()) {
+                continue;
+            }
+
+            $next = str_getcsv($next,"|");
+            $date = $next[0];
+            if ($whatstat == 0) {
+                $next = preg_grep("/wait/",$next);
+            } else if ($whatstat == 1) {
+                $next = preg_grep("/grade/",$next);
+            }
+            foreach ($next as $k=>$v) {
+                $next = explode(" ", $v);
+                foreach ($next as $value) {
+                    if ($value != 0) {
+                        $next = $value;
+                        break;
+                    }
+                }
+            }
+            if (gettype($next) != "String" && $next == array()) {
+                continue;
+            }
+            $next = floatval($next);
+            if($next != 0)
+                $array[] = array($date,$next);
+
+        }
+        return $array;
+    }
+
+    private function formatdate($date) {
```

Comment:
Use lowerCamelCase for function names (applies to other functions in this file as well)